server/plugin_test.go

```diff
@@ -373,3 +372,111 @@ func TestSyncUsers(t *testing.T) {
 		})
 	}
 }
+
+func TestConnectTeamsAppClient(t *testing.T) {
+	for _, test := range []struct {
+		Name          string
+		SetupAPI      func(*plugintest.API)
+		SetupClient   func(*mocks.Client)
+		ExpectedError string
+	}{
+		{
+			Name: "ConnectTeamsAppClient: Unable to connect to the app client",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogError", "Unable to connect to the app client", "error", mock.Anything).Times(1)
+			},
+			SetupClient: func(client *mocks.Client) {
+				client.On("Connect").Return(errors.New("unable to connect to the app client")).Times(1)
+			},
+			ExpectedError: "unable to connect to the app client",
+		},
+		{
+			Name: "ConnectTeamsAppClient: Valid",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogError", "Unable to connect to the app client", "error", mock.Anything).Times(1)
+			},
+			SetupClient: func(client *mocks.Client) {
+				client.On("Connect").Return(nil).Times(1)
+			},
+		},
+	} {
+		t.Run(test.Name, func(t *testing.T) {
+			assert := assert.New(t)
+			p := newTestPlugin()
+			test.SetupAPI(p.API.(*plugintest.API))
+			test.SetupClient(p.msteamsAppClient.(*mocks.Client))
+			err := p.connectTeamsAppClient()
+			if test.ExpectedError != "" {
+				assert.Contains(err.Error(), test.ExpectedError)
+			} else {
+				assert.Nil(err)
+			}
+		})
+	}
+}
+
+func TestStart(t *testing.T) {
+	for _, test := range []struct {
+		Name        string
+		SetupAPI    func(*plugintest.API)
+		SetupStore  func(*storemocks.Store)
+		SetupClient func(*mocks.Client)
+	}{
+		{
```

Comment:
Is there any reason to not have a success case here?

Reply:
a function `startSubscriptions` is called in this function which calls `p.clusterMutex.Lock()` which returns a `nil pointer dereference` error, due to which we are not able to add a success case here

Reply:
probably because the cluster mutex that you are creating is an empty cluster mutex. You can create a fake cluster mutex with something like `cluster.NewMutex(p.API, clusterMutexKey)`. That way, it should work fine.

Reply:
updated the test case